### PR TITLE
Update Content-Security-Policy header usage explanation

### LINKED ISSUE #33598: Improve Content-Security-Policy usage explanation
### What is the improvement or update you wish to see?

Can you add content security policy header examples?

### Is there any context that might help us understand?

At the moment the docs have
```
{
  key: 'Content-Security-Policy',
  value: // Your CSP Policy
}
```
Could you give examples of what `// Your CSP Policy` could be? 
Is it a string, an object, an array? 

In the Mozilla docs they have:
`<meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src https://*; child-src 'none';">`
but I'm not sure how to add that to `value: // Your CSP Policy`

For my use case I need to specify acceptable origins of our iframes.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/advanced-features/security-headers

### PR DESCRIPTION
**Pull request by amandeepmittal**
This PR improves the Content-Security-Policy header usage explanation in the `next.config.js` file.

## Bug

- [x] Related issues linked using fixes #33598 
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [x] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/advanced-features/security-headers.md [hunk 1]
```diff
@@ -113,10 +113,29 @@ This header helps prevent cross-site scripting (XSS), clickjacking and other cod
...
+
+When a directive uses a keyword such as `self`, wrap it in single quotes `''`.
+
+In the header's value, replace the new line with an empty string.
+
 ```jsx
 {
   key: 'Content-Security-Policy',
-  value: // Your CSP Policy
+  value: ContentSecurityPolicy.replace(/\n/g, '')
```
Comment (kachkaev) [inline docs/advanced-features/security-headers.md:138]:
```suggestion
  value: ContentSecurityPolicy.replace(/\s{2,}/g, ' ')
```

Comment (amandeepmittal) [reply docs/advanced-features/security-headers.md:138]: Thanks @kachkaev 🙏

Comment (kachkaev) [reply docs/advanced-features/security-headers.md]: Sorry looks like I forgot about the trailing newline! We also need `trim()`

```diff
-value: ContentSecurityPolicy.replace(/\s{2,}/g, ' ')
+value: ContentSecurityPolicy.replace(/\s{2,}/g, ' ').trim()
```

Comment (amandeepmittal) [reply docs/advanced-features/security-headers.md]: Thanks again! I'll update that.

**Review by styfle — APPROVED** [review]
I would usually have an array of directives and then use `arr.join('; ')` but this works too, thanks!

**Pull request merged**